ckeditor/plugins/a2j-guid/dialogs/guid.js

```diff
@@ -200,9 +201,20 @@ CKEDITOR.dialog.add('guid', function(editor) {
           html: '<ul class="list-group"></ul>',
 
           commit: function (widget) {
-            if (selectedVariable) {
-              widget.setData('guid', selectedVariable.getAttribute('guid'))
-              widget.setData('name', selectedVariable.getHtml())
+            var dialog = CKEDITOR.dialog.getCurrent()
+            var selectedElement = dialog.getSelectedElement()
+            var url = selectedResource.getAttribute('guid')
+            var name = selectedResource.getHtml()
+
+            if (selectedResource) {
+              widget.setData('guid', url)
+              widget.setData('name', name)
+              // update the original source element's html on commit
+              if (selectedElement) {
+                // child 0 is the legal-nav-resource custom element
+                var sourceElement = selectedElement.getChild(0)
```

Comment:
the 0th element is the legal-nav-resource-id and the 1 element is the span/wrapper that gives the little drag handle thing.